src/library/predefined-groups.c

```diff
@@ -0,0 +1,101 @@
+/*
+ * predefined-groups.c - loader for predefined groups
+ * Copyright (c) 2020 Red Hat Inc., Durham, North Carolina.
+ * All Rights Reserved.
+ *
+ * This software may be freely redistributed and/or modified under the
+ * terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2, or (at your option) any
+ * later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; see the file COPYING. If not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor
+ * Boston, MA 02110-1335, USA.
+ *
+ * Authors:
+ *   Radovan Sroka <rsroka@redhat.com>
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+
+#include <sys/stat.h>
+#include <fcntl.h>
+
+#include "attr-groups.h"
+#include "message.h"
+#include "string-util.h"
+
+#define SHELLS "/etc/shells"
+#define BUFFER_SIZE 1024
+
+static int load_shell_group(size_t index)
+{
+	int fd;
+	FILE * f = NULL;
+
+	char buf[BUFFER_SIZE];
+	memset(buf, 0, BUFFER_SIZE);
+
+	// Now open the file and load them one by one.
+	fd = open(SHELLS, O_NOFOLLOW|O_RDONLY);
```

Comment:
Rather than use /etc/shells which can be modified to anything, would it be better to just define the list of shells? Also, looking at the one on my system, I would not consider tmux to be a shell. And its missing csh, tcsh, zsh, ksh, ash, and fish.